### PR TITLE
fix(ui5-application-writer): Fixes incorrect deps written to manifest

### PR DESCRIPTION
**Pull request by IainSAP**
https://github.com/SAP/open-ux-tools/issues/308

Add ui5 dependencies to manifest as required. Fixes inconsistent default libs `sap.m` in ui5-local.yaml but `sap.ui.core` in manifest.

### CODE REVIEW

**Review by zdravko-georgiev — APPROVED** [review]
LGTM

File: packages/ui5-application-writer/src/data/defaultSettings.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { App, Package, UI5, UI5Framework } from '../types';
```
Comment (IainSAP) [inline packages/ui5-application-writer/src/data/defaultSettings.ts:1]: File rename is a temp workaround for sonar/lcov issues reporting the whole file as uncovered.

File: packages/ui5-application-writer/src/data/ui5Libs.ts [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+/**
+ * Merges the specified ui5 libs with the defaults.
+ * 
+ * @param ui5Libs - The ui5 libraries to be merged with the defaults
+ * @returns UI5 libs with defaults
+ */
 export const getUI5Libs = (ui5Libs?: string | string[]): string[] => {
     const libs = Array.isArray(ui5Libs) ? ui5Libs : ui5Libs?.split(',') || [];
-    return ['sap.m'].concat(libs).filter((value, index, self) => {
+    return defaultUI5Libs.concat(libs).filter((value, index, self) => {
```
Comment (ullasholla) [inline packages/ui5-application-writer/src/data/ui5Libs.ts:11]: Do we need to handle the case where the default libs are also passed in? We'll have duplicated entries then?

**Review by ullasholla — DISMISSED** [review]
Thanks @IainSAP, the fix looks good.
One question though: https://github.com/SAP/open-ux-tools/pull/309/files#r799688307

File: packages/ui5-application-writer/src/data/ui5Libs.ts (hunk 2 above)
Comment (IainSAP) [reply packages/ui5-application-writer/src/data/ui5Libs.ts:11]: That occurred to me also, but this filter condition removes duplicates.

**Review by longieirl — APPROVED** [review]

**Review by devinea — APPROVED** [review]
@IainSAP looks great. Thanks.

**Pull request merged**